src/main/java/com/sngular/kloadgen/sampler/schemaregistry/impl/ApicurioSchemaRegistry.java

```diff
@@ -30,147 +26,130 @@
-      List<SearchedArtifact> artifacts = this.schemaRegistryClient.searchArtifacts(null, null, null,
-                                                                                   null, null, null, null, null, null).getArtifacts();
+      final List<SearchedArtifact> artifacts = this.schemaRegistryClient.searchArtifacts(null, null, null,
+                                                                                         null, null, null, null, null, null).getArtifacts();
 
-      for (SearchedArtifact searchedArtifact : artifacts) {
-        subjects.add(searchedArtifact.getName());
+      for (final SearchedArtifact searchedArtifact : artifacts) {
+        subjects.add(searchedArtifact.getId());
       }
       return subjects;
-    } catch (RestClientException e) {
+    } catch (final RestClientException e) {
       throw new KLoadGenException(e.getMessage());
     }
   }
 
   @Override
-  public BaseSchemaMetadata<ApicurioSchemaMetadata> getLatestSchemaMetadata(String subjectName) throws KLoadGenException {
+  public final BaseSchemaMetadata<ApicurioSchemaMetadata> getLatestSchemaMetadata(final String artifactId) throws KLoadGenException {
     try {
-      final SearchedArtifact searchedArtifact = getLastestSearchedArtifact(subjectName);
+      final SearchedArtifact searchedArtifact = getLastestSearchedArtifact(artifactId);
       final ArtifactMetaData artifactMetaData = this.schemaRegistryClient.getArtifactMetaData(searchedArtifact.getGroupId(), searchedArtifact.getId());
-      return new BaseSchemaMetadata(new ApicurioSchemaMetadata(artifactMetaData));
-    } catch (RestClientException e) {
+      return new BaseSchemaMetadata<>(new ApicurioSchemaMetadata(artifactMetaData));
+    } catch (final RestClientException e) {
       throw new KLoadGenException(e.getMessage());
     }
   }
 
   @Override
-  public BaseParsedSchema<ApicurioParsedSchemaMetadata> getSchemaBySubject(String subjectName) {
+  public final BaseParsedSchema<ApicurioParsedSchemaMetadata> getSchemaBySubject(final String artifactId) {
     final ApicurioParsedSchemaMetadata schema = new ApicurioParsedSchemaMetadata();
     try {
-      List<SearchedArtifact> artifacts = this.schemaRegistryClient.searchArtifacts(null, subjectName, null,
-                                                                                   null, null, null, null, null, null).getArtifacts();
-      if (artifacts.isEmpty()) {
-        throw new KLoadGenException(String.format("Schema %s not found", subjectName));
+      final SearchedArtifact searchedArtifact = getLastestSearchedArtifact(artifactId);
+      final InputStream inputStream = this.schemaRegistryClient.getLatestArtifact(searchedArtifact.getGroupId(), searchedArtifact.getId());
+      final String result = IOUtils.toString(inputStream, String.valueOf(StandardCharsets.UTF_8));
+      final String searchedArtifactType = searchedArtifact.getType();
+      if (SchemaTypeEnum.AVRO.name().equalsIgnoreCase(searchedArtifactType)) {
```

Comment:
Maybe we can make a refactor here:
The first subjection is changing the `if` clause by a `switch`, or using another patron to avoid it.
Another point is to create a constant for `result.getBytes(StandardCharsets.UTF_8)` which is called three times (in each condition).